SCons/Script/Main.py

```diff
@@ -492,6 +492,16 @@ def GetOption(name):
 def SetOption(name, value):
     return OptionsParser.values.set_option(name, value)
 
+def ValidateOptions():
+    """
+    If you call this after you set all your command line options with AddOption(),
+    it will verify that all command line options are valid.
+    So if you added an option --xyz and you call SCons with --xyy you can cause
```

Comment:
Don't we have things like ninja tool which could add options later? The forced hard exit with failure seems a little limiting. Couldn't it just like return a true/false (or just raise an exception) so the user can then make further decisions on how to handle the situation?

Reply:
Looks like it shouldn't return, but throwing an exception is an option.
Here's action optparse's error()'s docstring. We override this in SCons' subclass.

```
        """error(msg : string)

        Print a usage message incorporating 'msg' to stderr and exit.
        If you override this in a subclass, it should not return -- it
        should either exit or raise an exception.
        """
```

If we throw a SCons specific exception, I can add the error string passed to `error(self, msg)` and the option parser to the exception so those could be retrieved.
Would that be sufficient for your use model?

What would you do in this case?

Reply:
most likely construct a more helpful and specific error message, like one about how our mongo `--variables-files` option is often misspelled, possibly depending on the option that is invalid, print a warning that it is being ignored and continue.